js/ui/collection/ui.collection_widget.edit.js

```diff
@@ -440,19 +440,30 @@ const CollectionWidget = BaseCollectionWidget.inherit({
     },
 
     _itemClickHandler: function(e) {
+        let itemSelectPromise;
+        const callBase = this.callBase;
+
         this._createAction((function(e) {
-            this._itemSelectHandler(e.event);
+            itemSelectPromise = this._itemSelectHandler(e.event);
         }).bind(this), {
             validatingTargetName: 'itemElement'
         })({
             itemElement: $(e.currentTarget),
             event: e
         });
 
-        this.callBase.apply(this, arguments);
+        if(itemSelectPromise) {
+            itemSelectPromise.always(() => {
+                callBase.apply(this, arguments);
+            });
+        } else {
+            callBase.apply(this, arguments);
+        }
     },
 
     _itemSelectHandler: function(e) {
+        let itemSelectPromise;
+
         if(!this.option('selectionByClick')) {
             return;
```

Comment:
maybe it's better to return `Promise.resolve()` here to simplify conditions in functions above?